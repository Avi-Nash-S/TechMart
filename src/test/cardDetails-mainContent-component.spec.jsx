import React from "react";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enzymeSerializer from "enzyme-to-json/serializer";
import CardDetailsMainContent from "../components/cardDetails-mainContent-component";
import { products } from "../static/test-data/constant-test-data";
expect.addSnapshotSerializer(enzymeSerializer);

configure({ adapter: new Adapter() });

describe("<CardDetailsMainContent/>", () => {
  it("Should match the component snapshot", () => {
    const wrapper = shallow(
      <CardDetailsMainContent currentProduct={products[0]} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render in DOM with correct Inputs", () => {
    const wrapper = mount(
      <CardDetailsMainContent currentProduct={products[0]} />
    );
    const container = wrapper.find(".card-details-main-container");
    expect(container.length).toEqual(1);
    const productPrice = wrapper.find(".price-info");
    expect(productPrice.length).toEqual(1);
  });
});
