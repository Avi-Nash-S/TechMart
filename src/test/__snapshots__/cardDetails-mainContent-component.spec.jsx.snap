// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardDetailsMainContent/> Should match the component snapshot 1`] = `
<div
  className="card-details-main-container"
>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      component="h2"
      gutterBottom={true}
      variant="h5"
    >
      SquareTrade 4-Year TV Protection Plan
      <span
        className="price-info"
      >
         Price: 
        $49.00
         
      </span>
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      color="textSecondary"
      component="h5"
      variant="body2"
    >
      <div
        className="review-container"
      >
        <Styled(MuiBox)
          borderColor="transparent"
          className="review-box"
          component="fieldset"
        >
          <WithStyles(ForwardRef(Rating))
            name="read-only"
            readOnly={true}
            value={0}
          />
        </Styled(MuiBox)>
        <div
          className="review-count"
        >
          Reviews:0
        </div>
      </div>
      <div
        className="long-description"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
        />
      </div>
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(CardContent))>
</div>
`;
